printer.go

```diff
@@ -104,8 +105,34 @@ func (p *printer) colorPrint(text, color string) {
 }
 
 func (p *printer) printString() {
+	quoted := strconv.Quote(p.value.String())
+	quoted = quoted[1 : len(quoted)-1]
+
 	p.colorPrint(`"`, "Red")
-	p.colorPrint(p.value.String(), "red")
+	for len(quoted) > 0 {
+		pos := strings.IndexByte(quoted, '\\')
+		if pos == -1 {
+			p.colorPrint(quoted, "red")
+			break
+		}
+		if pos != 0 {
+			p.colorPrint(quoted[0:pos], "red")
+		}
+
+		n := 1
+		switch quoted[pos+1] {
+		case 'x': // "\x00"
+			n = 3
+		case 'u': // "\u0000"
+			n = 5
+		case 'U': // "\U00000000"
+			n = 9
+		case '0', '1', '2', '3', '4', '5', '6', '7', '8', '9': // "\000"
```

Comment:
By the way, go compiler's scanner does not accept this type of literal whose number starts with '8' or '9'. [★](https://github.com/golang/go/blob/b7bfb54eaa84a7de187df905ae8f74c8fed515f4/src/text/scanner/scanner.go#L469-L470)
So you don't have to care about '8' and '9' .

And current `strconv.Quote()`'s implementation returns not `\000` but `\x00` for both "\000" and "\x00". [★](https://github.com/golang/go/blob/b7bfb54eaa84a7de187df905ae8f74c8fed515f4/src/strconv/quote.go#L63-L66)
Maybe L130-L131 are not used.